File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/controller/EntityCtrl.java
package com.itangcent.springboot.demo.controller;


import com.itangcent.springboot.demo.common.dto.Result;
import com.itangcent.springboot.demo.common.model.UserInfo;
import org.springframework.http.HttpEntity;
import org.springframework.http.RequestEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


/**
 * Spring Entity Test Controller
 * Demonstrates the usage of various Entity objects provided by Spring
 * Including RequestEntity, ResponseEntity, and HttpEntity
 */
@RestController
@RequestMapping(value = "/test/entity")
public class EntityCtrl extends BaseController {

    /**
     * Integer type ResponseEntity test
     * Demonstrates how to use ResponseEntity to wrap primitive type response
     *
     * @param token Token from request header
     * @param requestEntity Request entity containing integer
     * @return Response entity containing integer
     * @public Marked as public interface
     */
    @RequestMapping("/int")
    public ResponseEntity<Integer> intInEntity(
            @RequestHeader("token") String token,
            @RequestBody RequestEntity<Integer> requestEntity) {
        return ResponseEntity.ok(65536);
    }

    /**
     * Non-generic ResponseEntity test
     * Demonstrates usage of ResponseEntity without specifying generic type
     *
     * @return Response entity containing string
     * @public Marked as public interface
     */
    @RequestMapping("/nil")
    public ResponseEntity nilEntity() {
        return ResponseEntity.ok("nil");
    }

    /**
     * Complex object ResponseEntity test
     * Demonstrates ResponseEntity wrapping Result<Double> object
     *
     * @param token Token from request header
     * @param requestEntity Request entity containing user information
     * @return Result response entity containing double value
     * @public Marked as public interface
     */
    @RequestMapping("/double")
    public ResponseEntity<Result<Double>> doubleInEntity(
            @RequestHeader("token") String token,
            @RequestBody RequestEntity<UserInfo> requestEntity) {
        return ResponseEntity.ok(Result.success(65536.0));
    }

    /**
     * HttpEntity test
     * Demonstrates HttpEntity wrapping user information response
     *
     * @param token Token from request header
     * @return Result response entity containing user information
     */
    @RequestMapping("/user")
    public HttpEntity<Result<UserInfo>> userInEntity(
            @RequestHeader("token") String token) {
        UserInfo userInfo = new UserInfo();
        userInfo.setAge(18);
        return ResponseEntity.ok(Result.success(userInfo));
    }

}

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/EntityCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import com.itangcent.common.model.UserInfo
import org.springframework.http.HttpEntity
import org.springframework.http.RequestEntity
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * Entity响应控制器
 * 用于演示Spring框架中各种Entity类型的使用方法
 */
@RestController
@RequestMapping(value = ["/test/entity"])
class EntityCtrl {

    /**
     * 测试ResponseEntity包装整数类型
     * 演示如何使用ResponseEntity返回简单的整数类型
     *
     * @param token 请求头中的token
     * @param requestEntity 包含整数的请求实体
     * @return 返回包装在ResponseEntity中的整数
     */
    @RequestMapping("/int")
    fun intInEntity(
            @RequestHeader("token") token: String,
            @RequestBody requestEntity: RequestEntity<Int>): ResponseEntity<Int> {
        return ResponseEntity.ok(65536)
    }

    /**
     * 测试ResponseEntity包装复合对象
     * 演示如何使用ResponseEntity返回包含Result的复合对象
     *
     * @param token 请求头中的token
     * @param requestEntity 包含用户信息的请求实体
     * @return 返回包装在ResponseEntity中的Result<Double>对象
     */
    @RequestMapping("/double")
    fun doubleInEntity(
            @RequestHeader("token") token: String,
            @RequestBody requestEntity: RequestEntity<UserInfo>): ResponseEntity<IResult<Double>> {
        return ResponseEntity.ok(Result.success(65536.0))
    }

    /**
     * 测试HttpEntity响应
     * 演示如何使用HttpEntity返回用户信息
     *
     * @param token 请求头中的token
     * @return 返回包装在HttpEntity中的用户信息
     */
    @RequestMapping("/user")
    fun userInEntity(
            @RequestHeader("token") token: String): HttpEntity<IResult<UserInfo>> {
        val userInfo = UserInfo()
        userInfo.age = 18
        return ResponseEntity.ok(Result.success(userInfo))
    }
}
